ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/ToolAdminDetailsWindow.ui.xml

```diff
@@ -0,0 +1,87 @@
+                                        allowBlank="false"
+                                        ui:field="nameEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <form:FieldLabel text="{appearance.toolImportDescriptionLabel}">
+                    <form:widget>
+                        <form:TextArea readOnly="false"
+                                       enabled="true"
+                                       ui:field="descriptionEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <form:FieldLabel ui:field="typeLabel">
+                    <form:widget>
+                        <form:TextField readOnly="true"
+                                        enabled="false"
+                                        ui:field="typeEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <form:FieldLabel text="{appearance.toolImportAttributionLabel}">
+                    <form:widget>
+                        <form:TextField readOnly="false"
+                                        enabled="true"
+                                        ui:field="attributionEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <form:FieldLabel text="{appearance.toolImportVersionLabel}">
+                    <form:widget>
+                        <form:TextField readOnly="false"
+                                        enabled="true"
+                                        ui:field="versionEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <form:FieldLabel ui:field="locationLabel">
+                    <form:widget>
+                        <form:TextField readOnly="false"
+                                        enabled="true"
+                                        allowBlank="false"
+                                        ui:field="locationEditor"/>
+                    </form:widget>
+                </form:FieldLabel>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <d:ToolImplementationEditor ui:field="implementationEditor"/>
+            </container:child>
+            <container:child layoutData="{layoutData}">
+                <d:ToolContainerEditor ui:field="containerEditor"/>
+            </container:child>
+        </container:VerticalLayoutContainer>
+    </g:FormPanel>
+</ui:UiBinder> 
```

Comment:
@ash3rz You can configure IntelliJ to always include a newline at the end of files.
Preferences > Editor > General, then check `Ensure line feed at file end on Save`
You can setup "Strip trailing spaces on Save" for Modified Lines there too.